### ISSUE TITLE
Use the free version (rather than the Entreprise Edition)

### ISSUE THREAD

**Issue by kemenaran** [YunoHost-Apps/mattermost_ynh#108]
Currently the package downloads the tarball for the Entreprise Edition, and not the free Community Edition.

Feature-wise, this is stricly the same (Entreprise Edition features are not unlocked unless a serial key is provided). But this creates a licensing problem, as the EE doesn't have a free licence (and the CE does).

- [ ] Fix the URLs and provide the CE instead
- [x] Ensure that migration from EE to CE works properly

**Comment by madmaxlamenace** [comment]
I forked the yunohost package and proceed to update from my forked package.
I don't see any problems after the update and i have the team edition when i read the "about" menu

**Comment by kemenaran** [comment]
@madmaxlamenace Thanks for testing the migration path, and confirming it works properly. I just opened #115 for this.

However, when thinking again about it, I wonder what will happen to potential Yunohost users that actually *did* activate a license for the Enterprise features: the package will be downgraded, and the Enterprise features won't be available to them anymore. I don't know if there are any of these users though.

Furthermore, if someone setup Mattermost using Yunohost, and later wants to enable Enterprise features while still using the Yunohost package, it won't be possible anymore (as only the Team Edition will be distributed).

To recap:

**Pros:**

- Distributing a free-software-licensed release, better aligned with Yunohost values.

**Cons:**

- Users can no longer enable Entreprise features.

**Comment by lapineige** [comment]
What about adding an option during the installation ? (and by default, the free version).

During the upgrade, is it possible to detect which version is used ? Then it would be possible to upgrade correctly  ?

**Comment by kemenaran** [comment]
> What about adding an option during the installation ? (and by default, the free version).

Good question. I just checked the code: it uses the `ynh_setup_source` helper to download the packaged release defined in `config/app.src`.

To give the choice to the user, the trick would be to create _two_ src files (e.g. `app-team.src` and `app-entreprise.src`). The scripts could then tell at runtime which file should be used for downloading the release.

It should even be possible to switch from one release to another, by flipping the package setting:

```shell
# Switch to the free version
sudo yunohost app setting mattermost "use_enterprise_version" -v "false"
sudo yunohost app upgrade mattermost

# Switch to the enterprise version
sudo yunohost app setting mattermost "use_enterprise_version" -v "true"
sudo yunohost app upgrade mattermost
```

The only downside is that would add more code, moreover on a path that wouldn't be used often and thus prone to silent breakage. I'm not sure how to write a good automated test for this, but if I find out this option  would definitely be the best choice.

**Comment by madmaxlamenace** [comment]
Maybe the use of the entreprise edition needs to show the licence to the user.

**Comment by kemenaran** [comment]
It is now possible to install either the free or the entreprise edition.
